### PR TITLE
Remove assert that AppCheckListener is always registered

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This fixes a regression in https://github.com/firebase/firebase-ios-sdk/pull/8898

### CODE REVIEW

**Review by paulb777 — APPROVED** [review]
LGTM to merge once at least one of the previously failing tests succeeds.

File: Firestore/core/src/remote/grpc_root_certificates_generated.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #ifndef FIRESTORE_CORE_SRC_REMOTE_GRPC_ROOT_CERTIFICATES_GENERATED_H_
 #define FIRESTORE_CORE_SRC_REMOTE_GRPC_ROOT_CERTIFICATES_GENERATED_H_
 
-#include <stddef.h>
+#include <cstdlib>
```
Comment (paulb777) [inline Firestore/core/src/remote/grpc_root_certificates_generated.h:6]: Is this intended?

Comment (schmidt-sebastian) [reply Firestore/core/src/remote/grpc_root_certificates_generated.h]: Reverted

**Comment by schmidt-sebastian** [comment]
FYI There still seems to be a test failure related to the previous PR. Will look in 30 minutes.

**Comment by maksymmalyhin** [comment]
@schmidt-sebastian @paulb777 I would be more comfortable to go for release only once we have comprehensive code coverage for the cases when App Check or/and Auth are not installed in an app.

**Review by maksymmalyhin — APPROVED** [review]
LGTM on green checks. Tests for missing Auth and/or App Check are preferable but are not a blocker if you prefer manual testing before a release.

**Pull request merged**